chain/beacon/drand/drand.go

```diff
@@ -201,11 +204,32 @@ func (db *DrandBeacon) VerifyEntry(curr types.BeaconEntry, prev types.BeaconEntr
 	return err
 }
 
-func (db *DrandBeacon) MaxBeaconRoundForEpoch(filEpoch abi.ChainEpoch) uint64 {
+func (db *DrandBeacon) MaxBeaconRoundForEpoch(nv network.Version, filEpoch abi.ChainEpoch) uint64 {
 	// TODO: sometimes the genesis time for filecoin is zero and this goes negative
 	latestTs := ((uint64(filEpoch) * db.filRoundTime) + db.filGenTime) - db.filRoundTime
+
+	if nv <= network.Version13 {
+		return db.maxBeaconRoundV1(latestTs)
+	}
+
+	return db.maxBeaconRoundV2(latestTs)
+}
+
+func (db *DrandBeacon) maxBeaconRoundV1(latestTs uint64) uint64 {
 	dround := (latestTs - db.drandGenTime) / uint64(db.interval.Seconds())
 	return dround
 }
 
+func (db *DrandBeacon) maxBeaconRoundV2(latestTs uint64) uint64 {
```

Comment:
While the function is correct, I'm simply suggesting that it may make sense to simply take the one from drand implementation ? https://github.com/drand/drand/blob/master/chain/time.go#L52 but no worries